schema/sale_artworks.js

```diff
@@ -0,0 +1,42 @@
+import gravity from "lib/loaders/legacy/gravity"
+import SaleArtwork from "./sale_artwork"
+import { parseRelayOptions } from "lib/helpers"
+import { pageable } from "relay-cursor-paging"
+import { connectionDefinitions, connectionFromArraySlice } from "graphql-relay"
+import { filterSaleArtworksArgs } from "schema/filter_sale_artworks"
+import { SaleArtworkAggregations, SaleArtworkCounts } from "schema/filter_sale_artworks"
+
+const DEFAULTS = {
+  aggregations: ["total"],
+  page: 1,
+  size: 10,
+  offset: 0,
+}
+
+const SaleArtworksType = connectionDefinitions({
+  name: "SaleArtworks",
+  nodeType: SaleArtwork.type,
+  connectionFields: {
+    aggregations: SaleArtworkAggregations,
+    counts: SaleArtworkCounts,
+  },
+}).connectionType
+
+export default {
+  args: pageable(filterSaleArtworksArgs),
+  description: "Sale Artworks Elastic Search results",
+  type: SaleArtworksType,
+  resolve: async (_root, options, _request, { rootValue: { accessToken } }) => {
+    const params = parseRelayOptions({ ...DEFAULTS, ...options })
+    const response = await gravity.with(accessToken)("filter/sale_artworks", params)
```

Comment:
I should have noticed this in your previous PR, sorry about that; can you make this a new style “authenticated” data loader? Should also remove need to rewire from the tests.

Reply:
Updated. That `rootValue: { someLoader }` pattern is nice! Took the hell right out of my tests, too, as the `sinon.withArgs().returns` mock flow is very inflexible.